File: lib/entities/Match.dart
import 'Team.dart';

class Match {
  final String id;
  final Team teamA;
  final Team teamB;
  final DateTime date;

  Match({
    required this.id,
    required this.teamA,
    required this.teamB,
    required this.date,
  });

  factory Match.fromJson(Map<String, dynamic> json) {
    return Match(
      id: json['_id'] as String,
      teamA: Team.fromJson(json['teamA']),
      teamB: Team.fromJson(json['teamB']),
      date: DateTime.parse(json['date']),
    );
  }
}

File: FlutterNodeBackend/models/Match.js
import mongoose from "mongoose";

const { Schema, model } = mongoose;

const matchSchema = new Schema({
    teamA: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Team', // References the Team model
        required: true
    },
    teamB: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Team', // References the Team model
        required: true
    },
    date: { 
        type: Date, 
        required: true // Ensures a match date is always provided
    }
}, { timestamps: true }); // Automatically add createdAt and updatedAt fields

export default model("Match", matchSchema);
